packages/mui-material/src/Modal/useModal.ts

```diff
@@ -113,7 +120,11 @@ function useModal(parameters: UseModalParameters): UseModalReturnValue {
     manager.remove(getModal(), ariaHiddenProp);
   }, [ariaHiddenProp, manager]);
 
-  React.useEffect(() => {
+  // We need useLayoutEffect to make sure
+  // aria-hidden tags have time to get cleaned up properly
+  // in handleClose->manager.remove->ariaHiddenElements
+  // in the case someone unmounts the Modal higher up the tree
+  React.useLayoutEffect(() => {
```

Comment:
Are you sure layout effect is absolutely necessary here?

Reply:
Added a test for it in Modal.test.js.

Reply:
I used our util `useEnhancedEffect` instead which handles on SSR as well.